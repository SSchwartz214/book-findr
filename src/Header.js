import React from "react";

const Header = () => {
  return (
    <header>
      <i class="fas fa-book-reader fa-2x" />
      <h1>BookFindr</h1>
    </header>
  );
};

export default Header;
