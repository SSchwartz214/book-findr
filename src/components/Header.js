import React from "react";

const Header = () => {
  return (
    <header onClick={() => window.open("/", "_self")}>
      <i className="fas fa-book-reader fa-2x" />
      <h1>BookFindr</h1>
    </header>
  );
};

export default Header;
