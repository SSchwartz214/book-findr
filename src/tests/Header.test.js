import React from "react";
import { shallow, mount } from "enzyme";
import Header from "../components/Header";

describe("Header", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Header />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
