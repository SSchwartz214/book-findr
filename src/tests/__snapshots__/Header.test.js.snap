// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <header>
        <i
          class="fas fa-book-reader fa-2x"
        />
        <h1>
          BookFindr
        </h1>
      </header>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="fas fa-book-reader fa-2x"
          />,
          <h1>
            BookFindr
          </h1>,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <i
            class="fas fa-book-reader fa-2x"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-book-reader fa-2x",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "i",
        },
        Object {
          "instance": <h1>
            BookFindr
          </h1>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BookFindr",
          },
          "ref": null,
          "rendered": Array [
            "BookFindr",
          ],
          "type": "h1",
        },
      ],
      "type": "header",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <header>
          <i
            class="fas fa-book-reader fa-2x"
          />
          <h1>
            BookFindr
          </h1>
        </header>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-book-reader fa-2x"
            />,
            <h1>
              BookFindr
            </h1>,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <i
              class="fas fa-book-reader fa-2x"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-book-reader fa-2x",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "i",
          },
          Object {
            "instance": <h1>
              BookFindr
            </h1>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BookFindr",
            },
            "ref": null,
            "rendered": Array [
              "BookFindr",
            ],
            "type": "h1",
          },
        ],
        "type": "header",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
